lib/config/sources/yaml_source.rb

```diff
@@ -12,7 +12,11 @@ def initialize(path)
 
       # returns a config hash from the YML file
       def load
-        result = YAML.load(ERB.new(IO.read(@path)).result) if @path and File.exist?(@path)
+        if @path and File.exist?(@path)
+          file_contents = IO.read(@path)
+          file_contents = ERB.new(file_contents).result if Config.evaluate_erb_in_yaml
```

Comment:
Minor suggestion that would be really nice: give users the option to opt into- or out of this behavior on a per-file basis if they're using the `YAMLSource` API directly. For example,

```ruby
Config.add_source!(Config::Sources::YAMLSource.new(path1)) # defaults to global Config.evaluate_erb_in_yaml
Config.add_source!(Config::Sources::YAMLSource.new(path1, evaluate_erb: false)) # overrides global setting
Config.add_source!(Config::Sources::YAMLSource.new(path1, evaluate_erb: true)) # overrides global setting
```

That way it's not all-or-nothing. We do something similar for `EnvSource` here: configuration is derived from the global values if left unspecified, but can always be overridden. 

https://github.com/rubyconfig/config/blob/618c839ccca454d419b0587f8a113e2fe6f3ebcd/lib/config/sources/env_source.rb#L10-L14

Reply:
Oh neat. Ok I'll spend some time with this. Feedback makes sense.